### ISSUE TITLE
[Tooltip] flashing when hidden under react 18, but no problem under react 17

### ISSUE THREAD

**Issue by gwsbhqt** [DouyinFE/semi-design#715]
### Which Component 出现bug的组件
  - Tooltip (at react 18)

### semi-ui version
  - latest (2.6.0)
    
### Expected result 期望的结果是什么
  - react 18 will release soon, hope fix the problem under react 18

### Actual result 实际的结果是什么
  - flashing when hidden with animation under react 18
  - no flashing when hidden without animation under react 18
  - no flashing when hidden with animation under react 17
  - no flashing when hidden without animation under react 17

https://user-images.githubusercontent.com/16471930/159643101-1b971058-d45e-42f5-8061-33ab7a72376a.mp4

### Steps to reproduce 复现步骤
  - [flashing under react 18](https://codesandbox.io/s/serene-butterfly-18t7yy?file=/src/App.js)
  - [no flashing under react 17](https://codesandbox.io/s/festive-ishizaka-byqz1m?file=/src/App.js)

thanks for your awesome work

**Comment by pointhalo** [comment]
Thanks for your feedback, the description is very detailed and clear👍. 
I believe this is caused by a modification I made in v2.1,  I'll follow up on this issue, but it's not expected to be resolved right away, maybe in 2-3 weeks.
I'm really sorry. In recent weeks, there have been many other arrangements at work, so the problem is dealt with slowly.

**Comment by pointhalo** [comment]
Update the current situation: 
I initially suspected that the logic in `@douyinfe/semi-animation-react` was affected by the react version. Recently, I tried to use the exclusion method to comment out a certain part of the code to locate the problem, but there was no obvious progress.
The problem seems a bit tricky and I need to spend some more time looking into it.

If there are other code snippets in doubt, please point them out.

**Comment by gwsbhqt** [comment]
It can be imagined that this problem is very difficult to solve, respect you

By the way, I used the vite plugin to do a hotfix: temporarily turned off all tooltip animations, can be a reference for others

![image](https://user-images.githubusercontent.com/16471930/170607386-42cf5b85-0242-414b-8b2f-d5b668e6f464.png)

**Comment by shijiatongxue** [comment]
this issue may relate with https://github.com/radix-ui/primitives/issues/1074

**Comment by pointhalo** [comment]
I also wrote a minimal reproduction demo.   
The main logic is as follows: the dom needs to have a fade-out animation, and it is hoped that the animation itself will be removed by listening to onAnimationEnd after the fade-out animation is completed.  
The performance in 17 and 18 will be significantly inconsistent, 18 will flash once after the animation is completed, before removing the node


online demo ：
- react 18：https://codesandbox.io/s/react-18-flash-3v5yb2?file=/src/App.js
- react 17：https://codesandbox.io/s/react1-16-17-normal-vbrw7i?file=/src/App.js

```jsx
import "./styles.css";
import React, { useState } from "react";

export const Transition = () => {
  const [transitionState, setT] = useState("");

  const [insert, setInsert] = useState(false);

  const handleLeave = () => {
    console.log("remove dom");
    setInsert(false);
  };

  const DOM = () => {
    const enterCls = `tooltip-bounceIn`;
    const leaveCls = `tooltip-zoomOut`;
    const animateStyle = {
      animationDirection: "normal",
      animationName: transitionState === "enter" ? enterCls : leaveCls,
      animationDuration: "1000ms"
    };

    const handleEnd = () => {
      if (transitionState === "enter") {
        console.log("animation end of enter");
      } else if (transitionState === "leave") {
        console.log("animation end of leave");
        handleLeave();
      }
    };
    return (
      <div style={{ ...animateStyle }} onAnimationEnd={handleEnd}>
        test
      </div>
    );
  };

  const toggleShow = (insert) => {
    if (!transitionState) {
      setT("enter");
      setInsert(insert);
    } else if (transitionState === "enter") {
      setT("leave");
    } else if (transitionState === "leave") {
      setT("enter");
      setInsert(insert);
    }
  };

  return (
    <>
      <div style={{ width: 200, height: 90, border: "1px solid pink" }}>
        {insert ? <DOM></DOM> : null}
      </div>
      <button onClick={() => toggleShow(true)}>show</button>
      <button onClick={() => toggleShow(false)}>hide</button>
    </>
  );
};

export default function App() {
  return (
    <div className="App">
      <Transition />
    </div>
  );
}

```

```css
@keyframes tooltip-bounceIn {
  from {
    opacity: 0;
    transform: scale(0.6, 0.6);
  }

  70% {
    opacity: 1;
    transform: scale(1.01, 1.01);
  }

  to {
    opacity: 1;
    transform: scale(1, 1);
    animation-timing-function: cubic-bezier(0.215, 0.61, 0.355, 1);
  }
}

@keyframes tooltip-zoomOut {
  from {
    opacity: 100%;
  }

  60% {
    opacity: 60%;
    transform: scale(0.6, 0.4);
  }

  to {
    opacity: 20%;
  }
}

```

**Comment by pointhalo** [comment]
Seems like that we have two ways to solve this problem
A. add `animation-fill-mode: 'forward' `
B. use `ReactDOM.flushSync` to wrap setState call which remove DOM after animation end

I prefer plan B. I tested locally under v16/v18. No other problems have been found so far. A beta fix is planned to be released today

**Comment by pointhalo** [comment]
fix @douyinfe/semi-ui@2.12.0-beta.5

**Comment by Saul-BT** [comment]
Hi, I have this same trouble with Next.js.

For now, I fixed it through global CSS (`animation-fill-mode: forwards;`):

```css
[class|="semi"] {
  animation-fill-mode: forwards;
}
```

But... ...I think that this is not the way.
